tests/keras_autolog/test_keras_autolog.py

```diff
@@ -53,7 +82,32 @@ def generator():
         else:
             model.fit(data, labels, epochs=10)
 
-    return client.get_run(run.info.run_id)
+        assert mlflow.active_run()
+        assert mlflow.active_run().info.run_id == run.info.run_id
+
+
+@pytest.fixture
+def keras_random_data_run(random_train_data, fit_variant, random_one_hot_labels, manual_run):
+    if manual_run:
+        mlflow.start_run()
+
+    mlflow.keras.autolog()
```

Comment:
Maybe wrap this in a `try-finally` block so that we always end the run? e.g.

```suggestion
try:
    mlflow.keras.autolog()
    data = random_train_data
    labels = random_one_hot_labels
    model = create_model()
    if fit_variant == 'fit_generator':
        def generator():
            while True:
                yield data, labels
        model.fit_generator(generator(), epochs=10, steps_per_epoch=1)
    else:
         model.fit(data, labels, epochs=10)
finally:
    mlflow.end_run()
```